lib/src/boilerplate_suggestors/boilerplate_utilities.dart

```diff
@@ -13,35 +13,86 @@
 // limitations under the License.
 
 import 'package:analyzer/dart/ast/ast.dart';
+import 'package:meta/meta.dart';
 import 'package:over_react_codemod/src/constants.dart';
 import 'package:over_react_codemod/src/util.dart';
 
 typedef void YieldPatch(
     int startingOffset, int endingOffset, String replacement);
 
+@visibleForTesting
+bool isPublicForTest = false;
+
 // Stub while <https://jira.atl.workiva.net/browse/CPLAT-9308> is in progress
-bool _isPublic(ClassDeclaration node) => false;
+bool isPublic(ClassDeclaration node) => isPublicForTest;
+
+/// Returns the annotation node associated with the provided [classNode]
+/// that matches the provided [annotationName], if one exists.
+AstNode getAnnotationNode(ClassDeclaration classNode, String annotationName) {
+  if (classNode.metadata.isEmpty) return null;
+
+  return classNode.metadata.singleWhere(
```

Comment:
#nit It's unlikely, but possible that there could be two annotations with the same `name`, which would throw a `StateError`.

e.g.,
```dart
// node.name.name is "Foo" for both of these
@Foo.bar() 
@Foo.baz()
class Quux {}
```
```dart
@deprecated
@deprecated
class WhoopsIAnnotatedThisTwiceOnAccident {}
```

Most usages of `singleWhere` in this PR—where it's not actually important that there's no greater than one matching item—should be changed to `firstWhere` to avoid throwing unexpectedly.